frontend/src/components/Navigator/index.js

```diff
@@ -420,81 +444,81 @@ function Navigator({
-                    { showSaveGoalsAndObjButton
-                      ? (
-                        <>
-                          <Button className="margin-right-1" type="button" onClick={onGoalFormNavigate}>{`Save ${isOtherEntityReport ? 'objectives' : 'goal'}`}</Button>
-                          <Button className="usa-button--outline" type="button" onClick={isOtherEntityReport ? onSaveDraftOetObjectives : onSaveDraftGoal}>Save draft</Button>
-                        </>
-                      ) : (
-                        <>
-                          <Button className="margin-right-1" type="button" onClick={onContinue}>Save and continue</Button>
-                          <Button className="usa-button--outline" type="button" onClick={onSaveDraft}>Save draft</Button>
-                        </>
+                && (
+                  <Container skipTopPadding>
+                    <NavigatorHeader
+                      key={page.label}
+                      label={page.label}
+                      titleOverride={page.titleOverride}
+                      formData={formData}
+                    />
+                    <Form
+                      className="smart-hub--form-large"
+                    >
+                      {page.render(
+                        additionalData,
+                        formData,
+                        reportId,
+                        onSaveDraftGoal,
+                        onSaveDraftOetObjectives,
                       )}
-
-                    {
-                      page.position <= 1
-                        ? null
-                        : <Button outline type="button" onClick={() => { onUpdatePage(page.position - 1); }}>Back</Button>
-                    }
-                  </div>
-                </Form>
-                <DismissingComponentWrapper
-                  shown={showSavedDraft}
-                  updateShown={updateShowSavedDraft}
-                  hideFromScreenReader={false}
-                >
-                  {lastSaveTime && (
-                  <Alert className="margin-top-3 maxw-mobile-lg" noIcon slim type="success" aria-live="off">
-                    Draft saved on
-                    {' '}
-                    {lastSaveTime.format('MM/DD/YYYY [at] h:mm a z')}
-                  </Alert>
-                  )}
-                </DismissingComponentWrapper>
-              </Container>
-            )}
+                      <div className="display-flex">
+                        {showSaveGoalsAndObjButton
+                          ? (
+                            <>
+                              <Button className="margin-right-1" type="button" disabled={isAppLoading} onClick={onGoalFormNavigate}>{`Save ${isOtherEntityReport ? 'objectives' : 'goal'}`}</Button>
+                              <Button className="usa-button--outline" type="button" disabled={isAppLoading} onClick={isOtherEntityReport ? onSaveDraftOetObjectives : onSaveDraftGoal}>Save draft</Button>
```

Comment:
I think the onClick handler should be more like:

```javascript
onClick={isOtherEntityReport ? () => onSaveDraftOetObjectives(false) : () => onSaveDraftGoal(false)}
```

because currently, `onSaveDraftGoal` will receive the click event for its `isAutoSave` argument which is truthy and so the overlay is not shown

![Screen Shot 2022-11-04 at 9 19 02 AM](https://user-images.githubusercontent.com/17074357/200011144-e3ba9dea-f868-4684-b7d0-551eb5c99dc0.png)

Reply:
Nice catch, will change.

Reply:
I have changed the code if you can have another look 👀